### PR TITLE
Migrate `servicetalk-client-api-internal` from jUnit4 to jUnit5

### PR DESCRIPTION
**Pull request by amitvc**
Motivation:

JUnit 5 leverages features from Java 8 or later, such as lambda functions, making tests more powerful and easier to maintain.
JUnit 5 has added some very useful new features for describing, organizing, and executing tests. For instance, tests get better display names and can be organized hierarchically.
JUnit 5 is organized into multiple libraries, so only the features you need are imported into your project. With build systems such as Maven and Gradle, including the right libraries is easy.
JUnit 5 can use more than one extension at a time, which JUnit 4 could not (only one runner could be used at a time). This means you can easily combine the Spring extension with other extensions (such as your own custom extension).
Modifications:

Unit tests have been migrated from JUnit 4 to JUnit 5
Result:

Module servicetalk-client-api-internal now runs tests using JUnit 5

### CODE REVIEW

File: servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerMultiTest.java [hunk 1]
```diff
@@ -34,8 +34,8 @@
 import static org.hamcrest.Matchers.is;
 
 public abstract class AbstractRequestConcurrencyControllerMultiTest {
-    @Rule
-    public final Timeout timeout = new ServiceTalkTestTimeout();
+    @RegisterExtension
+    public final ExecutorExtension<Executor> exec = ExecutorExtension.withCachedExecutor();
```
Comment (idelpivnitskiy) [inline servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerMultiTest.java:38]: Why this is necessary? `exec` is not used in this test class

File: servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerOnlySingleTest.java [hunk 2]
```diff
@@ -34,8 +34,8 @@
 import static org.hamcrest.Matchers.is;
 
 public abstract class AbstractRequestConcurrencyControllerOnlySingleTest {
-    @Rule
-    public final Timeout timeout = new ServiceTalkTestTimeout();
+    @RegisterExtension
+    public final ExecutorExtension<Executor> exec = ExecutorExtension.withCachedExecutor();
```
Comment (idelpivnitskiy) [inline servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerOnlySingleTest.java:38]: Same question

**Review by idelpivnitskiy — COMMENTED** [review]
Thanks for the PR @amitvc!

Please, make sure that the build runs correctly. Currently, `quality` task failed with:
```
Execution failed for task ':servicetalk-client-api-internal:pmdTest'.
> 35 PMD rule violations were found. See the report at: file:///home/runner/work/servicetalk/servicetalk/servicetalk-client-api-internal/build/reports/pmd/test.html
```

File: servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerMultiTest.java (hunk 1 above)
Comment (amitvc) [reply servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerMultiTest.java]: You are right. It is not needed. I removed it.

File: servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerOnlySingleTest.java (hunk 2 above)
Comment (amitvc) [reply servicetalk-client-api-internal/src/test/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyControllerOnlySingleTest.java]: I removed it.

**Comment by amitvc** [comment]
@idelpivnitskiy pushed another revision. Successfully ran ./gradlew --parallel clean quality locally as well as test.

**Comment by idelpivnitskiy** [comment]
The CI reports:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':servicetalk-client-api-internal:pmdTest'.
> 35 PMD rule violations were found. See the report at: file:///home/runner/work/servicetalk/servicetalk/servicetalk-client-api-internal/build/reports/pmd/test.html
```
This happens because CI always cherry-picks changes from PR to the up-to-date `main` branch. Right before your PR we added a new PMD rule for jUnit5 tests (see #1596).

Please, merge the `main` branch into yours or rebase and fix the PMD warnings.

**Comment by amitvc** [comment]
@idelpivnitskiy thanks for the review. I have merged the main branch into mine. Hopefully, the build will go through. I will work on other packages in the list from the original issue.

**Comment by idelpivnitskiy** [comment]
@amitvc it was not enough just to merge `main` into your branch. Merging just gives you up to date version of PMD rules, the same as the CI sees. Now, run:
```
./gradlew :servicetalk-client-api-internal:build
```
And make sure it succeeds.

**Comment by amitvc** [comment]
@idelpivnitskiy thanks for the hint. Indeed when I run on this locally it fails. I am fixing this now.

**Review by bondolo — APPROVED** [review]

**Comment by bondolo** [comment]
Thank you @amitvc !

**Pull request merged**